app/Helpers/StringHelpers.php

```diff
@@ -166,21 +166,36 @@ public static function truncateStr($string, $length = 150)
      * PUB-137: Split the last "word" from a text. Used to keep reference
      * numbers and backref arrows from being orphaned onto the next line.
      *
-     * Will *not* work with HTML! Plain text only.
+     * If the last word contains HTML, we can't use it because it would break
+     * that element. Instead, we can try to use any terminating punctuation as
+     * the "last word". If the last word contains HTML and there is no
+     * punctuation this will not return a last word.
      */
-    public static function getLastWord($originalText)
+    public static function getLastWord(string $originalText): array
     {
         $originalText = rtrim($originalText);
+        // Split on the last space (" ") character as long as the final set of
+        // characters does not contain angle brackets (likely an HTML element).
         preg_match('/(.* )([^<>]+)$$/', $originalText, $textParts);
+        $beforeLastWord = $textParts[1] ?? $originalText;
+        $lastWord = $textParts[2] ?? '';
+        // If the last word did contain an HTML elment, find the last non-space,
+        // non-word character (usually punctuation) and use that as the last
+        // "word".
+        if (empty($lastWord)) {
+            preg_match('/(.*)([\S\W])$/', $originalText, $textAndPunctuation);
```

Comment:
I was really trying to be clever and combine this with the regex above, but I couldn't figure it out. If anyone is a regex-fu master, I'd happily take refactor suggestions.

Reply:
It's more clear to read separate, so I say keep it as is.